### PR TITLE
Replicated SDK instance-tags API

### PR DESCRIPTION
**Pull request by FourSigma**
Add descriptions of the new `/app/instance-tags` endpoint on Replicated SDK

### CODE REVIEW

File: docs/reference/replicated-sdk-apis.md [hunk 1]
```diff
@@ -92,7 +92,39 @@ Response:
 
 ### POST /app/custom-metrics
 
-Send custom application metrics. For more information and examples see [Configuring Custom Metrics](/vendor/custom-metrics)
+Send custom application metrics. For more information and examples see [Configuring Custom Metrics](/vendor/custom-metrics).
```
Comment (ajp-io) [inline docs/reference/replicated-sdk-apis.md:95]: Unrelated to this PR, but we should have a request and response for this too, even if we have more detailed info elsewhere.

File: docs/reference/replicated-sdk-apis.md [hunk 2]
```diff
@@ -92,7 +92,38 @@ Response:
...
+
+### POST /app/instance-tags
+
+Programmatically set new instance tags or overwrite existing tags. Instance tags are key-value pairs, where the key and the value are strings.
+
+Setting a tag with the `name` key will set the instance's name in the vendor portal.
+
+The `force` parameter defaults to `false`. If `force` is `false`, pre-existing tags will not be overwritten and the existing tags take precedence. If the `force` parameter is set to `true`, any pre-existing tags will be overwritten.
```
Comment (ajp-io) [inline docs/reference/replicated-sdk-apis.md:103]: I simplified this a lot, but I think it still gets the point across. I found the old wording a bit confusing (e.g., how often is "periodically"?).

File: docs/reference/replicated-sdk-apis.md [hunk 3]
```diff
@@ -92,7 +92,38 @@ Response:
...
 ### POST /app/custom-metrics
 
-Send custom application metrics. For more information and examples see [Configuring Custom Metrics](/vendor/custom-metrics)
+Send custom application metrics. For more information and examples see [Configuring Custom Metrics](/vendor/custom-metrics).
+
+### POST /app/instance-tags
+
+Programmatically set new instance tags or overwrite existing tags. Instance tags are key-value pairs, where the key and the value are strings.
+
+Setting a tag with the `name` key will set the instance's name in the vendor portal.
```
Comment (ajp-io) [inline docs/reference/replicated-sdk-apis.md:101]: Added a note about this so people know how to set instance name

File: docs/reference/replicated-sdk-apis.md [hunk 4]
```diff
@@ -92,7 +92,38 @@ Response:
...
+
+```json
+{
+  "data": {
+    "force": "false",
+    "tags": {
+      "name": "my-instance-name",
+      "preExistingKey": "will-not-be-overwritten",
+      "cpuCores": "10",
+      "supportTier": "basic",
```
Comment (ajp-io) [inline docs/reference/replicated-sdk-apis.md:120]: Removed a few tags for simplicity and made `"preExistingKey": "will-not-be-overwritten"` to show it won't be overwritten.

**Review by cbodonnell — DISMISSED** [review]

**Comment by AmberAlston** [comment]
Minor update to better match modal wording

File: docs/reference/replicated-sdk-apis.md (hunk 4 above)
Comment (AmberAlston) [reply docs/reference/replicated-sdk-apis.md:120]: I'm leaning towards removing this `preExistingKey` example. I'm concerned readers will get confused they have to pass this to not have things overwritten. Seems silly that someone might read it that way, but we've had folks get confused in the past that "{my app}" doesn't work in our example cli/api commands.

**Review by AmberAlston — APPROVED** [review]

**Review by paigecalvert — APPROVED** [review]

**Pull request merged**